overlord/devicestate/devicestate.go

```diff
@@ -293,6 +293,26 @@ func CanManageRefreshes(st *state.State) bool {
 	return false
 }
 
+// extractDownloadInstallEdgesFromTs is a helper that extract the first, last
+// download phase and install phase tasks from a TaskSet
+func extractDownloadInstallEdgesFromTs(ts *state.TaskSet) (firstDl, lastDl, firstInst, lastInst *state.Task) {
+	edgeTask := ts.Edge(snapstate.DownloadAndChecksDoneEdge)
+	tasks := ts.Tasks()
+	// we know we always start with downloads
+	firstDl = tasks[0]
+	// and always end with installs
+	lastInst = tasks[len(tasks)-1]
+	for _, t := range tasks {
```

Comment:
Now that, I believe, I understand this loop, would it be more direct to code this as follows:
```
tasks := ts.Tasks()
var edgeTaskIndex int
for i, task := range tasks {
  if task == edgeTask {
    edgeTaskIndex = i
    break
  }
}
return tasks[0], tasks[edgeTaskIndex], tasks[edgeTaskIndex+1], tasks[len(tasks)-1]
```

Reply:
Sure, this works as well.